sources/packages/forms/src/form-definitions/educationprogramoffering.json

```diff
@@ -1756,525 +1784,858 @@
+              "persistent": false,
+              "hidden": false,
+              "clearOnHide": false,
+              "refreshOn": "",
+              "redrawOn": "",
+              "modalEdit": false,
+              "dataGridLabel": false,
+              "labelPosition": "top",
+              "description": "",
+              "errorLabel": "",
+              "tooltip": "",
+              "tabindex": "",
+              "disabled": false,
+              "autofocus": false,
+              "dbIndex": false,
+              "customDefaultValue": "",
+              "calculateValue": "",
+              "calculateServer": false,
+              "widget": null,
+              "attributes": {},
+              "validateOn": "change",
+              "validate": {
+                "required": false,
+                "custom": "",
+                "customPrivate": false,
+                "strictDateValidation": false,
+                "multiple": false,
+                "unique": false
+              },
+              "conditional": {
+                "show": null,
+                "when": null,
+                "eq": ""
+              },
+              "overlay": {
+                "style": "",
+                "left": "",
+                "top": "",
+                "width": "",
+                "height": ""
+              },
+              "allowCalculateOverride": false,
+              "encrypted": false,
+              "showCharCount": false,
+              "showWordCount": false,
+              "properties": {},
+              "allowMultipleMasks": false,
+              "addons": [],
+              "tree": false,
+              "lazyLoad": false,
+              "title": "Panel",
+              "theme": "default",
+              "breadcrumb": "default",
+              "id": "enysm7c"
+            },
             {
-              "attr": "",
-              "value": ""
+              "label": "ineligibleDaysForFundingAfter10PercentageCalculation",
+              "labelWidth": "",
+              "labelMargin": "",
+              "customClass": "",
+              "modalEdit": false,
+              "defaultValue": null,
+              "persistent": true,
+              "protected": false,
+              "dbIndex": false,
+              "encrypted": false,
+              "redrawOn": "",
+              "customDefaultValue": "",
+              "calculateValue": "// Calculate MAX(0,(SUM(TOTAL ELIGIBLE BREAK DAYS)- 0.1(TOTAL DAYS)).\r\n// i.e, sum of all TOTAL ELIGIBLE BREAK DAYS minus 10% of total study days.\r\n\r\n// Sum of all eligible break days.\r\nconst sumOfTotalEligibleBreakDays = data.sumOfTotalEligibleBreakDays;\r\n\r\n// Allowable 10% total days.\r\nconst allowable10Percentage = data.allowable10Percentage;\r\n\r\n// Calculating the ineligible days\r\nvalue =\r\n  typeof sumOfTotalEligibleBreakDays === typeof 1 &&\r\n  typeof allowable10Percentage === typeof 1\r\n    ? sumOfTotalEligibleBreakDays - allowable10Percentage\r\n    : 0;\r\n\r\n",

```

Comment:
Form.io allow us to use the `moment` library. Is it not possible to use the library to simplify all the dates manipulation?

![image](https://user-images.githubusercontent.com/61259237/177871389-772ff7ec-43ff-449e-85bd-0d4d8f5f4eae.png)

Reply:
sure, I will use moment, it will make it easy, I didn't think of it.

Reply:
I believe that we never used before @ann-aot and these changes are heavily based on date checks.
If you can make it work we can define it as the way to go. Same for Loadash, IMO.